python/grass/pygrass/vector/__init__.py

```diff
@@ -461,6 +462,7 @@ def viter(self, vtype, idonly=False):
                         c_mapinfo=self.c_mapinfo,
                         table=self.table,
                         writeable=self.writeable,
+                        is2D=is2D,
```

Comment:
The Geo class can do it automatically by using `is2D=None`, however, it is probably better to fix it in Geo class itself. It has `is2D=True` in parameters, but `None` is a more reasonable default especially given that it determines the 2D/3D automatically if not given.

Reply:
Yes, but passing `is2D=None` to the `Geo` class will force performing 3D check for every geometry in vector map. It's not expected that you have mixed 2D/3D geometries in single 3D vector map. That's reason why this PR sets `is2D` variable before looping over geometries.

Reply:
Changing `is2D` default value to `None` would make sense. But than it should be also changed in other parts of the code, eg. https://github.com/OSGeo/grass/blob/main/python/grass/pygrass/vector/geometry.py#L1928

Reply:
You are right, checking for every geometry wouldn't be good. `self.is_3D()` call is just better. No need to solve the Geo issue here, but the default definitively seems wrong.